### PR TITLE
feat(excludeFromSave): Added option to exclude files from organizeOnSave option

### LINKED ISSUE #400: Add option to ignore files from beeing auto-sorted
#### System information

- OS: Win10
- TypeScript Hero version: 2.1.1
- VSCode version: 1.19.3

#### Feature Option to ignore files from beeing auto-sorted

When organizeOnSave is enabled, all files are always auto-sorted. There are files which are structured in a specific way - often with alot of comments and those should not be sorted.

Public example and possibly even a good value for a default is the "polyfill.ts" which is generated by angular-cli

#### Reason

In order to apply changes to the polyfill.ts without changing it's whole strucutre, organizeOnSave needs to be disabled in settings, changes need to be applied and organizeOnSave needs to be re-enabled again.

### PR DESCRIPTION
**Pull request by pfeigl**
- Closes #400

Started work on this and is already working. Tests missing and also wanted to log away the actual rule which resulted into skipping the organizeOnSave.

Also need to update changelog and documentation.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -234,6 +234,18 @@
           "description": "Defines if the imports should be organized on save.",
           "scope": "resource"
         },
+        "typescriptHero.imports.excludeFromSave": {
```
Comment (buehler) [inline package.json]: I'd name it `excludeFromOrganizeOnSave` to clarify the use case

File: src/import-organizer/index.ts [hunk 2]
```diff
@@ -52,6 +52,13 @@ export default class ImportOrganizer implements Activatable {
           );
           return;
         }
+        if (this.config.imports.excludeFromSave(event.document.uri)
+          .some(regexString => new RegExp(regexString).test(event.document.fileName))) {
+          this.logger.debug(
+            '[ImportOrganizer] OrganizeOnSave is skipped through excludeFromSave for this file',
```
Comment (buehler) [inline src/import-organizer/index.ts]: also log the filepath :-)

**Comment by pfeigl** [comment]
Updated the pull request with the requested changes (improved logging and renaming of config option).
Also merged develop into my branch, because there were some significant changes to filenames / structure.

Haven't been able to execute the tests locally (should they run on windows?) - thus haven't been writing any tests now. 

Instead tried to test as good as possible while debugging the extension with 0, 1, and n existing exclude rules.

Here's also a sample output from the debug console:
```
2018-02-25T19:11:26.344Z - debug: [ImportOrganizer] OrganizeOnSave is skipped through excludeFromOrganizeOnSave for this file {"excludePattern":"polyfills\\.ts$","file":"c:\\Development\\docu-safe\\DocuSafe\\src\\DocuSafe.UI\\wwwroot\\polyfills.ts"}
```

**Comment by pfeigl** [comment]
Will you merge this in it's current state? Or do I have to provide the test coverage (which is a little hard for me, because I couldn't get the existing tests to run on my windows box)

**Comment by pfeigl** [comment]
@buehler *hinthint* ;-)

**Comment by konuch** [comment]
Bump

**Pull request closed without merging**